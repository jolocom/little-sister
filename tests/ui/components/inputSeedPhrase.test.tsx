import React from 'react'
import { shallow } from 'enzyme'
import InputSeedPhraseComponent from '../../../src/ui/recovery/components/inputSeedPhrase'

describe('inputSeedPhrase component', () => {
  const makeProps = (
    input: string = '',
    suggestions: string[] = [],
    isLoading: boolean = false,
  ) => ({
      handleSubmit: jest.fn(),
      handleCancel: jest.fn(),
      isLoading
  })

  it('matches the snapshot at start', () => {
    const props = makeProps()
    const rendered = shallow(<InputSeedPhraseComponent {...props} />)
    expect(rendered).toMatchSnapshot()
  })

  // it('matches the snapshot while inputting', () => {
  //   const props = makeProps('in', ['insert', 'input'])
  //   const rendered = shallow(<InputSeedPhraseComponent {...props} />)
  //   expect(rendered).toMatchSnapshot()
  // })

  it('matches the snapshot when loading', () => {
    const props = makeProps('', [], true)
    const rendered = shallow(<InputSeedPhraseComponent {...props} />)
    expect(rendered).toMatchSnapshot()
  })
})
