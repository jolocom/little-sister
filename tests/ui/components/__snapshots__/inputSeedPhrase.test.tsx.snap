// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputSeedPhrase component matches the snapshot at start 1`] = `
<Wrapper
  style={
    Array [
      Object {
        "backgroundColor": "rgb(5, 5, 13)",
        "flex": 1,
        "justifyContent": "flex-start",
      },
      false,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "rgb(255, 239, 223)",
        "fontFamily": "TT Commons",
        "fontSize": 30,
        "lineHeight": 38,
        "marginTop": 32,
      }
    }
  >
    Recovery
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": 130,
        "justifyContent": "center",
        "marginHorizontal": 32,
        "marginTop": 8,
      }
    }
    testID="seedPhraseMsg"
  >
    <Text
      style={
        Object {
          "color": "rgb(255, 239, 223)",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "lineHeight": 26,
          "textAlign": "center",
        }
      }
      testID="recoveryMsg"
    >
      Start writing your seed-phrase and it will appear here word by word
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginHorizontal": 32,
      }
    }
  >
    <View
      style={
        Object {
          "width": 30,
        }
      }
    />
    <View
      style={
        Object {
          "width": 29,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={false}
      onChangeText={[MockFunction]}
      onSubmitEditing={[MockFunction]}
      placeholder="Tap here"
      placeholderTextColor="rgba(255, 255, 255, 0.4)"
      rejectResponderTermination={true}
      returnKeyLabel="Done"
      returnKeyType="next"
      selectionColor="rgb(148, 47, 81)"
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 22,
            "paddingVertical": 7,
          },
          Object {
            "color": "white",
          },
        ]
      }
      testID="seedWordFld"
      textAlign="center"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "marginRight": 10,
          "width": 19,
        }
      }
    />
    <View
      style={
        Object {
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(255, 239, 223)",
          "height": 1,
          "marginHorizontal": 32,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "rgba(255, 255, 255, 0.8)",
        "fontSize": 16,
        "marginTop": 3,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
        "marginTop": 24,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "left": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
      testID="seedWordSuggestions"
    />
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginBottom": 30,
        "width": "100%",
      }
    }
  >
    <JolocomButton
      containerStyle={
        Object {
          "borderColor": "rgb(255, 239, 223)",
          "borderWidth": 1,
          "marginHorizontal": 30,
          "marginTop": 12,
        }
      }
      onPress={[MockFunction]}
      text="Back to signup"
      transparent={true}
    />
  </View>
</Wrapper>
`;

exports[`inputSeedPhrase component matches the snapshot when loading 1`] = `
<Wrapper
  style={
    Array [
      Object {
        "backgroundColor": "rgb(5, 5, 13)",
        "flex": 1,
        "justifyContent": "flex-start",
      },
      Object {
        "justifyContent": "space-around",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "rgb(255, 239, 223)",
        "fontFamily": "TT Commons",
        "fontSize": 30,
        "lineHeight": 38,
        "marginTop": 32,
      }
    }
  >
    Full phrase verification
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": 130,
        "justifyContent": "center",
        "marginHorizontal": 32,
        "marginTop": 8,
      }
    }
    testID="seedPhraseMsg"
  >
    <Text
      style={
        Object {
          "color": "rgb(255, 239, 223)",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "lineHeight": 26,
          "textAlign": "center",
        }
      }
      testID="recoveryMsg"
    >
      Start writing your seed-phrase and it will appear here word by word
    </Text>
  </View>
  <RippleLoader
    color="rgb(89, 167, 145)"
    size={120}
    strokeWidth={4}
  />
  <View
    style={
      Object {
        "margin": 20,
      }
    }
  />
</Wrapper>
`;

exports[`inputSeedPhrase component matches the snapshot while inputting 1`] = `
<Wrapper
  style={
    Array [
      Object {
        "backgroundColor": "rgb(5, 5, 13)",
        "flex": 1,
        "justifyContent": "flex-start",
      },
      false,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "rgb(255, 239, 223)",
        "fontFamily": "TT Commons",
        "fontSize": 30,
        "lineHeight": 38,
        "marginTop": 32,
      }
    }
  >
    Recovery
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": 130,
        "justifyContent": "center",
        "marginHorizontal": 32,
        "marginTop": 8,
      }
    }
    testID="seedPhraseMsg"
  >
    <Text
      style={
        Object {
          "color": "rgb(255, 239, 223)",
          "fontFamily": "TT Commons",
          "fontSize": 20,
          "lineHeight": 26,
          "textAlign": "center",
        }
      }
      testID="recoveryMsg"
    >
      Start writing your seed-phrase and it will appear here word by word
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginHorizontal": 32,
      }
    }
  >
    <View
      style={
        Object {
          "width": 30,
        }
      }
    >
      <PreviousIcon
        onPress={[MockFunction]}
      />
    </View>
    <View
      style={
        Object {
          "width": 29,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={false}
      onChangeText={[MockFunction]}
      onSubmitEditing={[MockFunction]}
      placeholder="Tap here"
      placeholderTextColor="rgba(255, 255, 255, 0.4)"
      rejectResponderTermination={true}
      returnKeyLabel="Done"
      returnKeyType="next"
      selectionColor="rgb(148, 47, 81)"
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 22,
            "paddingVertical": 7,
          },
          Object {
            "color": "white",
          },
        ]
      }
      testID="seedWordFld"
      textAlign="center"
      underlineColorAndroid="transparent"
      value="in"
    />
    <View
      style={
        Object {
          "marginRight": 10,
          "width": 19,
        }
      }
    />
    <View
      style={
        Object {
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(255, 239, 223)",
          "height": 1,
          "marginHorizontal": 32,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "rgba(255, 255, 255, 0.8)",
        "fontSize": 16,
        "marginTop": 3,
      }
    }
  >
    Choose the right word or press enter
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
        "marginTop": 24,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "left": 0,
          "position": "absolute",
          "width": "100%",
        }
      }
      testID="seedWordSuggestions"
    >
      <ThemedComponent
        key="0"
        onPress={[Function]}
        style={
          Object {
            "container": Object {
              "backgroundColor": "rgba(148, 47, 81, 0.4)",
              "borderRadius": 8,
              "height": 40,
              "margin": 4,
              "minWidth": 0,
            },
            "text": Object {
              "color": "rgb(255, 239, 223)",
              "fontSize": 20,
              "fontWeight": "normal",
              "paddingVertical": 12,
            },
          }
        }
        testID="seedSuggestion0"
        text="insert"
        upperCase={false}
      />
      <ThemedComponent
        key="1"
        onPress={[Function]}
        style={
          Object {
            "container": Object {
              "backgroundColor": "rgba(148, 47, 81, 0.4)",
              "borderRadius": 8,
              "height": 40,
              "margin": 4,
              "minWidth": 0,
            },
            "text": Object {
              "color": "rgb(255, 239, 223)",
              "fontSize": 20,
              "fontWeight": "normal",
              "paddingVertical": 12,
            },
          }
        }
        testID="seedSuggestion1"
        text="input"
        upperCase={false}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginBottom": 30,
        "width": "100%",
      }
    }
  >
    <JolocomButton
      containerStyle={
        Object {
          "borderColor": "rgb(255, 239, 223)",
          "borderWidth": 1,
          "marginHorizontal": 30,
          "marginTop": 12,
        }
      }
      onPress={[MockFunction]}
      text="Back to signup"
      transparent={true}
    />
  </View>
</Wrapper>
`;
